engine/glfw/lib/cocoa/cocoa_window.m

```diff
@@ -132,14 +132,39 @@ - (void)applicationDidUpdate:(NSNotification *)notification
     // Wait for the first update to make window inactive and then activate it again
     // It helps to avoid issue when the window opens inactive
     // https://github.com/defold/defold/issues/6709
-    if( !_glfwLibrary.Unbundled ) {
-        ProcessSerialNumber psn = { 0, kCurrentProcess };
-        TransformProcessType( &psn, kProcessTransformToBackgroundApplication );
-        [self performSelector:@selector(activateProcess) withObject:nil afterDelay:0.1];
-        _glfwLibrary.Unbundled = GL_TRUE;
+    if( _glfwLibrary.FirstUpdate )
+    {
+        // Starting from macOS Sonoma (14.0) it should be solved in two different ways depends 
+        // if it's bundled app or not
+        // https://github.com/defold/defold/issues/8066
+        NSOperatingSystemVersion version = [[NSProcessInfo processInfo] operatingSystemVersion];
+        if (version.majorVersion < 14)
+        {
+            [self deactivateProcess];
+        }
+        else
+        {
+            if ( _glfwLibrary.Unbundled )
+            {
+                [self deactivateProcess];
+            }
+            else
+            {
+                [self activateWindow];
+            }
```

Comment:
on newer macos it should be activated in two different ways depends on if it's bundled or not